xdsl/dialects/riscv.py

```diff
@@ -600,6 +609,59 @@ def assembly_line_args(self) -> tuple[AssemblyInstructionArg, ...]:
         return self.rd, self.rs1, self.rs2
 
 
+class RdRsRsFloatOperationWithFastMath(IRDLOperation, RISCVInstruction, ABC):
```

Comment:
Could this subclass `RdRsRsOperation[FloatRegisterType, FloatRegisterType, FloatRegisterType]`? Is there a problem with the attributes being passed in?

Reply:
OK, I'll change it and see if it complaints

Reply:
I remember it being annoying about something similar, but I'm not sure exactly what the condition was for it to fail. It might be easier to duplicate the classes for now, rather than deal with irdl logic

Reply:
With

```py
class RdRsRsFloatOperationWithFastMath(RdRsRsOperation[FloatRegisterType, FloatRegisterType, FloatRegisterType]):
```

the same `super().__init__` call complains that arguments are missing for `rs1`, `rs2` and `rd`.

Not sure what irks it.